specifyweb/frontend/js_src/lib/components/DataModel/uniquenessRules.ts

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+
+import { formsText } from '../../localization/forms';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import {
+  type GetOrSet,
+  type IR,
+  type RA,
+  type RR,
+  setDevelopmentGlobal,
+} from '../../utils/types';
+import { formatConjunction } from '../Atoms/Internationalization';
+import { load } from '../InitialContext';
+import { strictGetModel } from './schema';
+import type { LiteralField, Relationship } from './specifyField';
+import type { Tables } from './types';
+
+export type UniquenessRule = {
+  readonly id: number | null;
+  readonly fields: RA<string>;
+  readonly scopes: RA<string>;
+  readonly modelName: keyof Tables;
+  readonly isDatabaseConstraint: boolean;
+};
+
+export type UniquenessRules = Partial<
+  RR<
+    keyof Tables,
+    | RA<{
+        readonly rule: UniquenessRule;
+        readonly duplicates?: UniquenessRuleValidation;
+      }>
+    | undefined
+  >
+>;
+
+export type UniquenessRuleValidation = {
+  readonly totalDuplicates: number;
+  readonly fields: RA<{
+    readonly fields: IR<string>;
+    readonly duplicates: number;
+  }>;
+};
+
+export let uniquenessRules: UniquenessRules = {};
+
+export const fetchContext = f
+  .all({
+    schemaBase: import('./schemaBase').then(
+      async ({ fetchContext }) => fetchContext
+    ),
+    schema: import('./schema').then(async ({ fetchContext }) => fetchContext),
+  })
+  .then(async ({ schemaBase }) =>
+    load<UniquenessRules>(
+      `/businessrules/uniqueness_rules/${schemaBase.domainLevelIds.discipline}/`,
+      'application/json'
+    )
+  )
+  .then((data) =>
+    Object.fromEntries(
+      Object.entries(data).map(([lowercaseTableName, rules]) => [
+        // Convert all lowercase table names from backend to PascalCase
+        strictGetModel(lowercaseTableName).name,
+        rules,
+      ])
+    )
+  )
+  .then((data) => {
+    uniquenessRules = data;
+    setDevelopmentGlobal('_uniquenessRules', data);
+    return data;
+  });
+
+export function getUniquenessRules(): UniquenessRules | undefined;
+export function getUniquenessRules<TABLE_NAME extends keyof Tables>(
+  table: TABLE_NAME
+): UniquenessRules[TABLE_NAME] | undefined;
+export function getUniquenessRules<TABLE_NAME extends keyof Tables>(
+  table?: TABLE_NAME
```

Comment:
for clarity, please rename `table` to `tableName`

Reply:
Fixed in https://github.com/specify/specify7/pull/4215/commits/43aee5b97e52edca55b58e0cb2b76773754b510f